scalalib/test/src/mill/scalalib/HelloWorldTests.scala

```diff
@@ -45,11 +48,79 @@ object HelloWorldTests extends TestSuite {
   }
 
   object HelloWorldWithMain extends HelloBase {
-    object core extends HelloWorldModule{
-      def mainClass = Some("Main")
+    object core extends HelloWorldModuleWithMain
+  }
+
+  val akkaHttpDeps = Agg(ivy"com.typesafe.akka::akka-http:10.0.13")
+
+  object HelloWorldAkkaHttpAppend extends HelloBase {
+    object core extends HelloWorldModuleWithMain {
+      def ivyDeps = akkaHttpDeps
+
+      def assemblyRules = T { Seq(Assembly.Rule.Append("reference.conf")) }
     }
   }
 
+  object HelloWorldAkkaHttpExclude extends HelloBase {
+    object core extends HelloWorldModuleWithMain {
+      def ivyDeps = akkaHttpDeps
+
+      def assemblyRules = T { Seq(Assembly.Rule.Exclude("reference.conf")) }
+    }
+  }
+
+  object HelloWorldAkkaHttpAppendPattern extends HelloBase {
+    object core extends HelloWorldModuleWithMain {
+      def ivyDeps = akkaHttpDeps
+
+      def assemblyRules = T { Seq(Assembly.Rule.AppendPattern(".*.conf")) }
+    }
+  }
+
+  object HelloWorldAkkaHttpExcludePattern extends HelloBase {
+    object core extends HelloWorldModuleWithMain {
+      def ivyDeps = akkaHttpDeps
+
+      def assemblyRules = T { Seq(Assembly.Rule.ExcludePattern(".*.conf")) }
+    }
+  }
+
+  object HelloWorldMultiAppend extends HelloBase {
+    object core extends HelloWorldModuleWithMain {
+      def moduleDeps = Seq(model)
+
+      def assemblyRules = T { Seq(Assembly.Rule.Append("reference.conf")) }
+    }
+    object model extends HelloWorldModule
+  }
+
+  object HelloWorldMultiExclude extends HelloBase {
+    object core extends HelloWorldModuleWithMain {
+      def moduleDeps = Seq(model)
+
+      def assemblyRules = T { Seq(Assembly.Rule.Exclude("reference.conf")) }
```

Comment:
Could we have one unit test for the behavior of anchored regexes? e.g. making sure`^reference.conf` matches only `reference.conf` and not `foo/reference.conf`, and that `reference.conf$` matches only `foo/reference.conf` but not `foo/reference.confetti`